src/deepsparse/v2/operators/engine_operator.py

```diff
@@ -0,0 +1,118 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Any, Dict, List, Optional, Union
+
+from deepsparse import Context, Engine, MultiModelEngine, Scheduler
+from deepsparse.benchmark import ORTEngine
+from deepsparse.utils import join_engine_outputs, split_engine_inputs
+from deepsparse.v2.operators import Operator
+
+
+DEEPSPARSE_ENGINE = "deepsparse"
+ORT_ENGINE = "onnxruntime"
+
+SUPPORTED_PIPELINE_ENGINES = [DEEPSPARSE_ENGINE, ORT_ENGINE]
+
+__all__ = ["EngineOperator"]
+
+
+class EngineOperator(Operator):
```

Comment:
definitely should have a defined input/output schema here for routers and schedulers to take advantage of